platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -392,23 +387,50 @@ String formatAdvice(List<String> replacements) {
         }
     }
 
-    static final Pattern ID_CLEAN_PATTERN = Pattern.compile("[A-Za-z]+");
+    public static final char DASH = '-';
 
     public static String createDefaultDeprecationId(String... ids) {
         StringBuilder sb = new StringBuilder();
         for (String id : ids) {
-            if(id == null){
+            if (id == null) {
                 continue;
             }
-            Matcher matcher = ID_CLEAN_PATTERN.matcher(id);
-            while (matcher.find()) {
-                sb.append(TextUtil.toLowerCaseLocaleSafe(matcher.group()));
-                sb.append("-");
+            CharSequence cleanId = getToken(id);
+            if (cleanId.length() > 0) {
+                sb.append(cleanId);
+                sb.append(DASH);
             }
         }
-        return removeEnd(sb.toString(), "-");
+        removeTrailingDashes(sb);
+        return sb.toString();
     }
 
+    private static void removeTrailingDashes(StringBuilder sb) {
```

Comment:
Just checking, the strings we manipulate here can only have ascii chars. That's a requirement otherwise we should think about locale/codepoints/etc.

Reply:
I think it would be fine to skip the non-ascii jars for the category name.